### PR TITLE
Augment allele observation

### PR DESCRIPTION
**Pull request by jstjohn**
Adding read pairing information, and the position within the read of an allele into the AlleleObservation model.

### CODE REVIEW

**Comment by jstjohn** [comment]
These changes were the result of a discussion we had earlier today where we thought it might be useful to add a few more things into the AlleleObservation object to enable read offset bias type calculations. This might not matter if we decide to just pass reads into Mutect directly though, so this PR is not high priority unless you think you will be useful?

File: avocado-core/src/main/scala/org/bdgenomics/avocado/discovery/ReadExplorer.scala [hunk 1]
```diff
@@ -78,6 +79,8 @@ class ReadExplorer(referenceObservations: RDD[Observation]) extends Explorer wit
         quals(readPos),
         mapq,
         negativeStrand,
+        firstOfPair,
+        readPos,
```
Comment (fnothaft) [inline avocado-core/src/main/scala/org/bdgenomics/avocado/discovery/ReadExplorer.scala]: If you want to store just one field, you can store the first of pair as positive numbers and the second of pair as negative indices. This is a micro-optimization, but it is consistent with BQSRs nomenclature, IIRC.

**Comment by fnothaft** [comment]
Thanks @jstjohn! This looks good. I'm going to sort out why @AmplabJenkins is unhappy; we seem to be having a transient Jenkins issue today.

**Comment by shaneknapp** [comment]
jenkins, test this please

**Comment by shaneknapp** [comment]
through the powers of black magic, apparently jenkins is no longer flaky.

File: avocado-core/src/main/scala/org/bdgenomics/avocado/discovery/ReadExplorer.scala (hunk 1 above)
Comment (fnothaft) [inline avocado-core/src/main/scala/org/bdgenomics/avocado/discovery/ReadExplorer.scala]: I'd say though that it's easier to store it as you've done. We can always go back and change things if necessary.

**Comment by fnothaft** [comment]
Merged! Thanks @jstjohn. H/T @shaneknapp for the Jenkins black magic.

**Pull request merged**